sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -170,7 +178,9 @@ func (c *control) handleInstruction(ctx context.Context, req *fnpb.InstructionRe
 			}
 			log.Debugf(ctx, "Plan %v: %v", desc.GetId(), plan)
 
+			c.mu.Lock()
```

Comment:
I want to move desc.GetID() out of the lock. The intent of the lock management code was to have only operations that obviously could not fail (map lookups and deletes) so it's easy to reason about all the manual lock management.